R/plot_alignment_table.R

```diff
@@ -20,54 +20,58 @@
 #'
 #' plot_alignment_table(toy_data_alignment_table %>% filter(asset_class == "equity"))
 plot_alignment_table <- function(data) {
-  env <- list(data = substitute(data))
-  check_data_alignment_table(data, env = env)
-
-  size_lim <- c(min(data$perc_aum, na.rm = TRUE), max(data$perc_aum, na.rm = TRUE))
-  size_range <- c(4, 12)
-
-  if (nrow(data %>% filter(.data$sector == "power")) > 0) {
-    p_power <- plot_alignment_table_sector_stripe(data, "power", size_lim,
-      size_range,
-      ncol = 4
-    )
-  } else {
-    p_power <- patchwork::plot_spacer()
-  }
-
-  if (nrow(data %>% filter(.data$sector == "fossil_fuels")) > 0) {
-    p_fossil <- plot_alignment_table_sector_stripe(data, "fossil_fuels", size_lim,
-      size_range,
-      ncol = 3
-    )
-  } else {
-    p_fossil <- patchwork::plot_spacer()
-  }
-
-  if (nrow(data %>% filter(.data$sector == "automotive")) > 0) {
-    p_auto <- plot_alignment_table_sector_stripe(data, "automotive", size_lim,
-      size_range,
-      ncol = 3
-    )
+  stopifnot(is.data.frame(data))
+  if (nrow(data) > 0) {
+    env <- list(data = substitute(data))
+    check_data_alignment_table(data, env = env)
+
+    size_lim <- c(min(data$perc_aum, na.rm = TRUE), max(data$perc_aum, na.rm = TRUE))
+    size_range <- c(4, 12)
+  
+    if (nrow(data %>% filter(.data$sector == "power")) > 0) {
+      p_power <- plot_alignment_table_sector_stripe(data, "power", size_lim,
+        size_range,
+        ncol = 4
+      )
+    } else {
+      p_power <- patchwork::plot_spacer()
+    }
+  
+    if (nrow(data %>% filter(.data$sector == "fossil_fuels")) > 0) {
+      p_fossil <- plot_alignment_table_sector_stripe(data, "fossil_fuels", size_lim,
+        size_range,
+        ncol = 3
+      )
+    } else {
+      p_fossil <- patchwork::plot_spacer()
+    }
+  
+    if (nrow(data %>% filter(.data$sector == "automotive")) > 0) {
+      p_auto <- plot_alignment_table_sector_stripe(data, "automotive", size_lim,
+        size_range,
+        ncol = 3
+      )
+    } else {
+      p_auto <- patchwork::plot_spacer()
+    }
+  
+  
+    p_ylabel <- ggplot(data.frame(l = "Aligned scenario temperature", x = 1, y = 1)) +
+      geom_text(aes(x = .data$x, y = .data$y, label = .data$l), angle = 90, size = 7) +
+      theme_void() +
+      coord_cartesian(clip = "off")
+  
+    p_tech <- (p_power / p_fossil / p_auto) +
+      plot_layout(guides = "collect")
+  
+    p <- p_ylabel + p_tech + plot_layout(widths = c(1, 15))
   } else {
-    p_auto <- patchwork::plot_spacer()
+    p <- empty_plot_no_data_message()
```

Comment:
NIT: Could be good to add a warning message here, to indicate that the input data won't result in a plot. (Not so useful for TM, but could be helpful when manually chunking through the code)

Reply:
isn't it enough that the returned plot is a message that there is no data?

Reply:
Very good point!

Reply:
I'd suggest having the warning in `empty_plot_no_data_message()`. It's useful to have both the visual (plot) output, but also have a record in the logs since we don't always have access to the visual output